src/lib.rs

```diff
@@ -226,6 +247,71 @@ pub struct PointsBatch {
 }
 
 impl PointsBatch {
+    pub fn append(&mut self, other: &mut PointsBatch) -> std::result::Result<(), String> {
+        if self.position.is_empty() {
+            *self = other.split_off(0);
+        } else {
+            assert_eq!(self.attributes.len(), other.attributes.len());
+            self.position.append(&mut other.position);
+            let data_types: Vec<(String, AttributeDataType)> = self
+                .attributes
+                .iter()
+                .map(|(name, attrib)| (name.clone(), attrib.data_type()))
+                .collect();
```

Comment:
Why collect?

Reply:
This can't be done without collect, because then `self` would be borrowed as immutable and mutable at the same time.